src/plugins/discover/public/application/angular/discover.js

```diff
@@ -411,13 +411,15 @@ function discoverController($route, $scope, Promise) {
       index: $scope.indexPattern.id,
       interval: 'auto',
       filters: _.cloneDeep(persistentSearchSource.getOwnField('filter')),
+      hideUnmapped: false,
     };
     if (savedSearch.grid) {
       defaultState.grid = savedSearch.grid;
     }
     if (savedSearch.hideChart) {
       defaultState.hideChart = savedSearch.hideChart;
     }
+    defaultState.hideUnmapped = savedSearch.hideUnmapped;
```

Comment:
do you think it makes sense not to set `hideUnmapped` in default state unless it is explicitly set in `savedSearch`? This way it would also not become part of the URL unless the user changes the switch. Think the currently functional failure would be fixes without editing any `expect`, and we would not add a variable to URL without user action.

Reply:
Yes, I think that's reasonable.